File: 242-valid-anagram/valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
       unordered_map<char, int> mp1;

       for(auto x:s){
        mp1[x]++;
       } 
       for(auto x:t){
        mp1[x]--;
        if(mp1[x]==0){
            mp1.erase(x);
        }
       }
       return mp1.size()==0;
    }
};

File: 242-valid-anagram/valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length()) return false;

        HashMap<Character, Integer> hp1 = new HashMap<>();
        HashMap<Character, Integer> hp2 = new HashMap<>();

        for (char c : s.toCharArray()) {
            hp1.put(c, hp1.getOrDefault(c, 0) + 1);
        }

        for (char c : t.toCharArray()) {
            hp2.put(c, hp2.getOrDefault(c, 0) + 1);
        }

        return hp1.equals(hp2);
    }
}
